vars/checkReleaseNotes.groovy

```diff
@@ -0,0 +1,118 @@
+ * @param Map args = [:] args A map of the following parameters
+ * @param args.version <required> - Release version. eg: 3.0.0
+ * @param args.dataTable <required> - Markdown data table file in the format: https://github.com/opensearch-project/opensearch-build/issues/3747#issuecomment-2704366641 eg: ./table.md
+ * @param args.action <optional> - Action to perform. Default is 'check'. Acceptable values are 'check' and 'notify'.
+ */
+void call(Map args = [:]) {
+    String action = args.action ?: 'check'
+    // Parameter check
+    validateParameters(args, action)
+    def version = args.version.tokenize('-')[0]
+
+    def parsedContent = new ParseReleaseNotesMarkdownTable(readFile(args.dataTable)).parseReleaseNotesMarkdownTableRows()
+    def componentsWithFalseStatus = parsedContent.findAll { it['Status'] == 'False' }
+    echo("Components missing release notes: " + componentsWithFalseStatus.collect { it['Component'] })
+
+    if (action == 'notify' && !componentsWithFalseStatus.isEmpty()) {
+        echo "Notifying all the components with missing release notes."
+        notifyReleaseOwners(version, componentsWithFalseStatus)
+    }
+}
+
+/**
+ * Validates input parameters
+ */
+private void validateParameters(Map args, String action) {
+    if (!args.version) {
+        error "Version parameter is required."
+    }
+
+    if (!args.dataTable || args.dataTable.isEmpty()) {
+        error "dataTable is required to get the content."
+    } else {
+        if (!fileExists(args.dataTable)) {
+            error("Invalid path.Data Table file does not exist at ${args.dataTable}")
+        }
+    }
+
+    List<String> validActions = ['check', 'notify']
+    if (!validActions.contains(action)) {
+        error "Invalid action '${action}'. Valid values: ${validActions.join(', ')}"
+    }
+}
+
+/**
+ * Notify components regarding the missing release notes by adding a comment to the release issue.
+ * @param version: Release version.
+ * @param componentsWithFalseStatus: Parsed content with status as False.
+ */
+private void notifyReleaseOwners(String version,def componentsWithFalseStatus) {
+    componentsWithFalseStatus.each { component ->
+        withCredentials([
+                string(credentialsId: 'jenkins-health-metrics-account-number', variable: 'METRICS_HOST_ACCOUNT'),
+                string(credentialsId: 'jenkins-health-metrics-cluster-endpoint', variable: 'METRICS_HOST_URL')]) {
+            withAWS(role: 'OpenSearchJenkinsAccessRole', roleAccount: "${METRICS_HOST_ACCOUNT}", duration: 900, roleSessionName: 'jenkins-session') {
+                def metricsUrl = env.METRICS_HOST_URL
+                def awsAccessKey = env.AWS_ACCESS_KEY_ID
+                def awsSecretKey = env.AWS_SECRET_ACCESS_KEY
+                def awsSessionToken = env.AWS_SESSION_TOKEN
+
+                ReleaseMetricsData releaseMetricsData = new ReleaseMetricsData(metricsUrl, awsAccessKey, awsSecretKey, awsSessionToken, version, this)
```

Comment:
These credential usages are universal for all different plugins right? Any harm we extract them outside the loop to avoid multiple times calling?
e.g. 
```
withCredentials([...] {
   withAWS(...) {
       ...
       ReleaseMetricsData releaseMetricsData = new ReleaseMetricsData(...)
       componentsWithFalseStatus.each { component ->
            ...
        }
   }
}

Reply:
They are universal but passed down in each call. If the withCrerdentials are only available within the block AFAIK. And to make those calls `getReleaseIssue` we need those credentials..

Reply:
Yeah. Since the entire function relies on credentials, I'm thinking we could put all related operations within a single credential block. This would prevent redundant calls for each component. Would it be possible to do it this way?

Reply:
We should not. Only use credentials where required. In case we want to add more functionality to component with false status, that should go outside credential block but within loop.
For example:
```
    componentsWithFalseStatus.each { component ->
        withCredentials([
                ........
                ........
                    addComment(releaseIssueUrl, ghCommentContent)
                }
         // do something else with these components here that has nothing to do with metrics credentials
            }
        }